script/versions.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python
+"""
+Query the github API for the git tags of a project, and return a list of
+version tags for recent releases, or the default release.
+
+The default release is the most recent non-RC version.
+
+Recent is a list of unqiue major.minor versions, where each is the most
+recent version in the series.
+
+For example, if the list of versions is:
+
+    1.8.0-rc2
+    1.8.0-rc1
+    1.7.1
+    1.7.0
+    1.7.0-rc1
+    1.6.2
+    1.6.1
+
+`default` would return `1.7.1` and
+`recent -n 3` would return `1.8.0-rc2 1.7.1 1.6.2`
+"""
+from __future__ import print_function
+
+import argparse
+import itertools
+import operator
+from collections import namedtuple
+
+import requests
+
+
+GITHUB_API = 'https://api.github.com/repos'
+
+
+class Version(namedtuple('_Version', 'major minor patch rc')):
+
+    @classmethod
+    def parse(cls, version):
+        version = version.lstrip('v')
+        version, _, rc = version.partition('-')
+        major, minor, patch = version.split('.', 3)
+        return cls(int(major), int(minor), int(patch), rc)
+
+    @property
+    def major_minor(self):
+        return self.major, self.minor
+
+    @property
+    def order(self):
+        return self[:3] + (None, self.rc) if self.rc else ('', )
```

Comment:
Yeah this is a bit of a puzzle to me, especially as I don't know the operator precedence between `+` and `if` off the top of my head.

Furthermore, I'd feel better if we didn't require readers of the code to know that fact about sorting. Instead of `None` and `''`, how about 0 and 1?

```python
if self.rc:
    return (self.major, self.minor, self.patch, 0, self.rc)
else:
    return (self.major, self.minor, self.patch, 1)
```

I know it's verbose, but at least it's really clear what the return value will look like in both cases, so I don't have to construct a tuple in my head and consult the class definition when reading.